storm-core/src/jvm/backtype/storm/utils/StormCommandExecutor.java

```diff
@@ -0,0 +1,785 @@
+                ArrayList<String>(), extraPaths, args, false, false, "");
+        return;
+    }
+
+    void devzookeeperCommand (List<String> args) {
+        System.out.println("Called devzookeeperCommand using reflection");
+        System.out.println("Arguments are : ");
+        for (String s: args) {
+            System.out.println(s);
+        }
+        return;
+    }
+
+    void versionCommand (List<String> args) {
+        System.out.println("Called versionCommand using reflection");
+        System.out.println("Arguments are : ");
+        for (String s: args) {
+            System.out.println(s);
+        }
+        return;
+    }
+
+    void monitorCommand (List<String> args) {
+        System.out.println("Called monitorCommand using reflection");
+        System.out.println("Arguments are : ");
+        for (String s: args) {
+            System.out.println(s);
+        }
+        return;
+    }
+
+    void uploadcredentialsCommand (List<String> args) {
+        System.out.println("Called uploadcredentialsCommand using reflection");
+        System.out.println("Arguments are : ");
+        for (String s: args) {
+            System.out.println(s);
+        }
+        return;
+    }
+
+    void geterrorsCommand (List<String> args) {
+        System.out.println("Called geterrorsCommand using reflection");
+        System.out.println("Arguments are : ");
+        for (String s: args) {
+            System.out.println(s);
+        }
+        return;
+    }
+
+    String getLog4jConfigDirectory () {
+        List<String> extraPaths = new ArrayList<String>();
+        extraPaths.add(this.clusterConfDirectory);
+        String log4jDirectory = this.confValue("storm.logback.conf.dir",
+                extraPaths, true);
+        if ((log4jDirectory == null) || ("".equals(log4jDirectory)) ||
+                ("nil".equals(log4jDirectory))) {
+            log4jDirectory = this.stormLog4jConfDirectory;
+        }
+        return log4jDirectory;
+    }
+
+    private void printUsage () {
+        String commands = StringUtils.join(this.COMMANDS, "\n\t");
+        System.out.println("Commands:\n\t" + commands);
+        System.out.println("\nHelp: \n\thelp \n\thelp <command>\n");
+        System.out.println("Documentation for the storm client can be found" +
+                " at "  +
+                "http://storm.incubator.apache" +
+                ".org/documentation/Command-line-client.html\n");
+        System.out.println("Configs can be overridden using one or more -c " +
+                "flags, e.g. " +
+                "\"storm list -c nimbus.host=nimbus.mycompany.com\"\n");
+    }
+
+    private void executeHelpCommand () {
+        System.out.println("Print storm help here");
+    }
+
+}
+
+class WindowsStormCommandExecutor extends StormCommandExecutor {
+
+    WindowsStormCommandExecutor () {
+
+    }
+
+    void initialize () {
+        return;
+    }
+
+    void execute (String[] args) {
```

Comment:
@caofangkun thanks for checking this out. The code is still not complete. I need to put the windows implementation as well. Most of the code in the UnixStormCommandExecutor will move to abstract base class so that WindowsStormCommandLineExecutor can reuse. It will only have code that is specific to windows.